examples/text-replacer/content.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+// Replace text on the page using a static list of patterns
+function textReplacer(replacements) {
+  const replacementPatterns = buildReplacementRegex(replacements);
+  replaceText(replacementPatterns);
+}
+
+function buildReplacementRegex(source) {
+  const output = [];
+  for (var i = 0; i < source.length; i++) {
+    if (!source[i]) { continue; }
+    const [find, replace] = source[i];
+    const sanitizedMatch = escapeRegExp(find);
+    const findExp = new RegExp(`\\b${sanitizedMatch}\\b`, 'gi');
+    output[i] = [findExp, replace];
+  }
+  return output;
+}
+
+// Use var to avoid "Identifier 'REGEXP_SPECIAL_CHARACTERS' has already been
```

Comment:
Oh wow. I didn't realise this runs non-ESM code and runs in the top-scope.

Everything in this file is being redeclared always. Would it be worth putting it in an IIFE? Is that too complex?

Reply:
+1